hack/docs/internal/cronjob-tutorial/writing_tests_controller.go

```diff
@@ -194,6 +191,9 @@ var _ = Describe("CronJob controller", func() {
 			controllerRef := metav1.NewControllerRef(createdCronjob, gvk)
 			testJob.SetOwnerReferences([]metav1.OwnerReference{*controllerRef})
 			Expect(k8sClient.Create(ctx, testJob)).Should(Succeed())
+			// Note that you can't set the status while creating the resource. It needs to be updated after
+			testJob.Status.Active = 2
```

Comment:
Also. we can do that in a follow up. It does not need to be in the PR we need to change this example in order to promote the good practices. Following the k8s API convention we should use status conditions, see this example: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4-with-deploy-image/api/v1alpha1/busybox_types.go#L41-L52

Then, see here in the controller: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4-with-deploy-image/internal/controller/busybox_controller.go#L44-L50

https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4-with-deploy-image/internal/controller/busybox_controller.go#L101-L119

So, we could update the cronjob tutorial and this example to work with StatusConditions instead.

Reply:
Just for you know :-)

Reply:
Good point. That should probably be a followup though since it'll require a bigger refactor